File: 03_question_marks/question_marks.js
function QuestionsMarks(str) {
  let num1 = null
  let countMarks = 0
  for(let char of str) {
    if (!isNaN(char)) {
      if (num1 !== null && parseInt(char) + num1 === 10 && countMarks === 3) {
        return true
      }
      num1 = parseInt(char)
      countMarks = 0
    } else if (char === '?') {
      countMarks++
    }
  }
  return false
}

const inputStr = "arrb6???4xxbl5???eee5";
const output = QuestionsMarks(inputStr);
console.log(output);

File: 03_question_marks/question_marks.py
def QuestionsMarks(str):
  num1 = None
  count_marks = 0

  for char in str:
    if char.isdigit():
      if num1 is not None and int(char) + num1 == 10 and count_marks == 3:
        return True
      num1 = int(char)
      count_marks = 0
    elif char == "?":
      count_marks += 1
  return False

input_str = 'arrb6???4xxbl5???eee5'
output = QuestionsMarks(input_str)
print(output)
